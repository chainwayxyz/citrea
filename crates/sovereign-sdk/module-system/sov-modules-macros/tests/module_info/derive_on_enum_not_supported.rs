use sov_modules_api::{ModuleInfo, StateMap};

#[derive(ModuleInfo)]
enum TestStruct<C: sov_modules_api::Context> {
    #[state]
    TestState1(StateMap<String, String>),

    #[state]
    TestState2(C),
}

fn main() {}
