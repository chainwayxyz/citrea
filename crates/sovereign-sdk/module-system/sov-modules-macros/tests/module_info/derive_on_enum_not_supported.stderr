error: The `ModuleInfo` macro supports structs only.
 --> tests/module_info/derive_on_enum_not_supported.rs:4:1
  |
4 | enum TestStruct<C: sov_modules_api::Context> {
  | ^^^^
